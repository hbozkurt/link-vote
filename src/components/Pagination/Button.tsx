import * as React from "react";

interface Props {
  active?: boolean;
  onClick?(): void;
  disabled?: boolean;
  classNames?: string;
}

export const Button: React.FC<Props> = props => {
  const cs = ["border", "w-8", "h-8", "p-4", "flex", "justify-center", "items-center"];
  cs.push(props.active ? "bg-gray-900 text-white cursor-default" : "cursor-pointer");
  
  if (props.disabled) {
    cs.push("cursor-not-allowed");
  }
  if (props.classNames) {
    cs.push(props.classNames);
  }

  return (
    <span
      className={cs.join(" ")}
      onClick={!props.disabled ? props.onClick : undefined}>
      <span>{props.children}</span>
    </span>
  );
}
