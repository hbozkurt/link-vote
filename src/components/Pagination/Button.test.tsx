import * as React from "react";
import { create } from "react-test-renderer";
import { Button } from "./Button";

describe("Button component", () => {
  test("Matches the snapshot", () => {
    const button = create(<Button />);
    expect(button.toJSON()).toMatchSnapshot();
  });
});
