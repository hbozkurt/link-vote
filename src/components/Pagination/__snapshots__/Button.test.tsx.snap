// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component Matches the snapshot 1`] = `
<span
  className="border w-8 h-8 p-4 flex justify-center items-center cursor-pointer"
>
  <span />
</span>
`;
